ietf-optical-impairment-topology.tree

```diff
@@ -49,39 +56,110 @@ module: ietf-optical-impairment-topology
     |  +--ro OTSiG-container
     |     +--ro OTSi* [OTSi-carrier-id]
     |        +--ro OTSi-carrier-id           int16
+    |        +--ro transponder-ref?          leafref
+    |        +--ro transceiver-ref?          leafref
+    |        +--ro configured-mode?          leafref
     |        +--ro OTSi-carrier-frequency?   decimal64
     |        +--ro OTSi-signal-width?        decimal64
     |        +--ro channel-delta-power?      decimal64
-    +--ro transponders-list* [transponder-id]
-       +--ro transponder-id                      uint32
-       +--ro (mode)?
-       |  +--:(G.692.2)
-       |  |  +--ro standard-mode?                standard-mode
-       |  +--:(organizational-mode)
-       |  |  +--ro operational-mode?             operational-mode
-       |  |  +--ro organization-identifier?      vendor-identifier
-       |  +--:(explicit-mode)
-       |     +--ro available-modulation-types*   identityref
-       |     +--ro configured-modulation-type?   identityref
-       |     +--ro available-baud-rates*         uint32
-       |     +--ro configured-baud-rate?         uint32
-       |     +--ro available-FEC-types*          identityref
-       |     +--ro configured-FEC-type?          identityref
-       |     +--ro FEC-code-rate?                decimal64
-       |     +--ro FEC-threshold?                decimal64
-       +--ro power?                              int32
-       +--ro power-min?                          int32
-       +--ro power-max?                          int32
-  augment /nw:networks/nw:network/nw:node/tet:te/tet:tunnel-termination-point:
-    +--ro transponder-list* [carrier-id]
+    |        +--ro central-frequency?        frequency-thz
+    |        +--ro channel-output-power?     dbm-t
+    |        +--ro channel-input-power?      dbm-t
+    |        +--ro total-input-power?        dbm-t
+    +--ro transponder-list* [transponder-id]
+       +--ro transponder-id      uint32
+       +--ro transceiver-list* [transceiver-id]
+          +--ro transceiver-id     uint32
+          +--ro supported-modes
+             +--ro supported-mode* [mode-id]
+                +--ro mode-id                      string
+                +--ro (mode)
+                   +--:(G.698.2)
+                   |  +--ro standard-mode?         standard-mode
+                   +--:(organizational-mode)
+                   |  +--ro organizational-mode
+                   |     +--ro operational-mode?
+                   |     |       operational-mode
+                   |     +--ro organization-identifier?
+                   |     |       organization-identifier
+                   |     +--ro min-central-frequency?
+                   |     |       frequency-thz
+                   |     +--ro max-central-frequency?
+                   |     |       frequency-thz
+                   |     +--ro minimum-channel-spacing?
+                   |     |       frequency-ghz
+                   |     +--ro min-channel-output-power?   dbm-t
```

Comment:
I am not so sure about all these capabilities listed in the oprational mode: does this mean that if the operational mode has already some defined values, these new ones overwite ? or does this mean that the operational mode may not have such info so we ensure that we can have what is needed ? in such case then these values might be common for all cases (standard, operational and explicit)
besides the same info might be needed out of the stantard mode as well.
Maybe I missed/forgot a decision point on this during a meeting?

Reply:
The contentious issue was whether the mode is a list (option-2) or a choice (option-1) (see https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/files/5104243/if-mode-clarification_options-200820-github.pptx ). 
The view taken by the majority was that Appplication-codes and Organizational-codes fully describe the 
Transponder capabilities and consequently no (or very few) additional capabilities of the type you pointed out would be required. As a consequence, the capabilities are explicit for the explicit mode and implicit for the other modes.
Probably best to discuss this with the proponents of the choice-structure https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/issues/29#issuecomment-702148542